src/entities.rs

```diff
@@ -29,17 +31,21 @@ impl Entity {
     /// `World::spawn_at` for easy serialization. Alternatively, consider `id` for more compact
     /// representation.
     pub fn to_bits(self) -> u64 {
-        u64::from(self.generation) << 32 | u64::from(self.id)
+        NonZeroU64::from(self.generation).get() << 32 | u64::from(self.id)
     }
 
     /// Reconstruct an `Entity` previously destructured with `to_bits`
     ///
     /// Useful for storing entity IDs externally, or in conjunction with `Entity::to_bits` and
     /// `World::spawn_at` for easy serialization.
-    pub fn from_bits(bits: u64) -> Self {
-        Self {
-            generation: (bits >> 32) as u32,
-            id: bits as u32,
+    pub fn from_bits(bits: u64) -> Option<Self> {
+        if NonZeroU32::new((bits >> 32) as u32).is_some() {
+            Some(Self {
+                generation: NonZeroU32::new((bits >> 32) as u32).unwrap(),
+                id: bits as u32,
+            })
+        } else {
+            None
         }
```

Comment:
It's redundant to call `new` twice.
```suggestion
        Some(Self {
            generation: NonZeroU32::new((bits >> 32) as u32)?
            id: bits as u32,
        })
```
You could also use `map`.